src/chains/filecoin/filecoin/src/ipfs-server.ts

```diff
@@ -29,16 +28,11 @@ class IPFSServer {
     this.httpServer = null;
   }
 
-  async start() {
-    // Uses a temp folder for now.
-    const folder: string = await new Promise((resolve, reject) => {
-      fs.mkdtemp(path.join(os.tmpdir(), "ganache-ipfs-"), (err, folder) => {
-        if (err) {
-          return reject(err);
-        }
-        resolve(folder);
-      });
-    });
+  async start(parentDirectory: string) {
+    const folder = path.join(parentDirectory, "ganache-ipfs");
+    if (!fs.existsSync(folder)) {
+      await fs.promises.mkdir(folder);
```

Comment:
I haven't followed how `parentDirectory` gets here - does this need the `{recursive: true}` in options?

Reply:
It comes from https://github.com/trufflesuite/ganache-core/pull/848/files#diff-a8d7ab65cce14da36b4292d6f60e99a3d13c4eb75c33cdd6f62d88cefbcb7e9aR194 (`blockchain.ts:194`). `parentDirectory` is the `Blockchain.#database.directory` and is set after initialization. So the directory should exist (unless the user has deleted it after initialization (which would cause several other issue, if even possible)

So `{ recursive: true }` is not necessary, but can be added if you would prefer it. There shouldn't be any valid scenario (currently) that would need it.